### PR TITLE
ADAPT-5093 | @rebeccahongsf | adjust utm params and update tsgg branch

### PR DESCRIPTION
**Pull request by rebeccahongsf**
# READY FOR REVIEW

# Summary
- adjust utm params
- update tsgg branch

# Review By (Date)
- asap

# Review Tasks

## Setup tasks and/or behavior to test

1. Navigate to https://deploy-preview-396--stanford-alumni.netlify.app?utm_source=test&utm_medium=test&olc=123456&utm_content=test
2. Log in as a test user
3. Navigate to Travel/Study by using the main navigation
4. Search for a trip in which you can register for (Italy Extension)
5. Navigate to that trip through the search results
6. Click on the reserve button to go to the interstitial page 
7. Add some people and continue to the registration form
8. Land on registration for with URL parameters still intact

# Associated Issues and/or People
- ADAPT-5093

### CODE REVIEW

File: src/api/auth/login.js [hunk 1]
```diff
@@ -5,10 +5,14 @@ import { authInstance } from '../../utilities/authInstance';
 const handler = connect()
   // Return user to page they were previously on after logging in.
   .use((req, res, next) => {
+    const { referer } = req.headers;
```
Comment (sherakama) [inline src/api/auth/login.js:8]: Referer is an unreliable source for information and we should avoid using it where possible.

File: src/api/auth/login.js [hunk 2]
```diff
@@ -5,10 +5,14 @@ import { authInstance } from '../../utilities/authInstance';
...
+    const { query, pathname } = URLParse(referer);
     if (!req.query.final_destination) {
-      const { referer } = req.headers;
-      const { pathname } = URLParse(referer);
-      req.query.final_destination = pathname;
+      req.query.final_destination = pathname || '/';
+    }
+    if (query.includes('utm') || query.includes('olc')) {
+      const urlParams = pathname + query;
+      req.query.final_destination = urlParams;
```
Comment (sherakama) [inline src/api/auth/login.js:15]: We shouldn't have to touch this file at all as the final_destination should be set on the login button. 

The example is in online giving https://github.com/SU-SWS/ood_online_giving/blob/dev/src/components/auth/loginButton.js#L29

and you can adjust the code in this repo here instead: https://github.com/SU-SWS/saa_alumni/blob/feature/ADAPT-5093--utm-params/src/components/navigation/accountLinks.js#L30

File: src/components/auth/AuthenticatedPage.js [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
 const AuthenticatedPage = ({ children, redirectUnauthorized = true }) => {
+  const location = useLocation();
+
   function redirectUnuauthorized() {
-    const returnUrl = window.location.pathname;
+    const returnUrl =
+      typeof window === 'undefined'
+        ? location.pathname
+        : window.location.pathname;
+    console.log('Return URL: ', returnUrl);
```
Comment (sherakama) [inline src/components/auth/AuthenticatedPage.js:14]: Remove console.log and add all search parameters to the return path

File: src/api/auth/login.js (hunk 2 above)
Comment (sherakama) [reply src/api/auth/login.js:15]: We also shouldn't bother checking for specific parameters here and just preserve everything.

File: src/components/cards/TripCard/TripCard.jsx [hunk 4]
```diff
@@ -61,7 +61,10 @@ const TripCard = ({
           <Date startDate={startDate} endDate={endDate} isSmall />
         </div>
         <div className={styles.content}>
-          <Link to={tripURL} className={styles.link}>
+          <SbLink
```
Comment (sherakama) [inline src/components/cards/TripCard/TripCard.jsx]: Woot thanks!

File: src/components/navigation/accountLinks.js [hunk 5]
```diff
@@ -22,13 +23,12 @@ const Initial = ({ string }) => {
...
 const AccountLinks = ({ mainLinkClasses }) => {
+  const location = useLocation();
   const ref = useRef(null);
   const buttonRef = useRef(null);
   const [expanded, setExpanded] = useState(false);
-  const loginDestination =
-    typeof window !== 'undefined' ? window.location.pathname : null;
   const loginParams = new URLSearchParams({
-    final_destination: loginDestination,
+    final_destination: location.pathname,
```
Comment (sherakama) [inline src/components/navigation/accountLinks.js:31]: add search parameters as well. They might need to be URL encoded to survive the round trip.

Comment (rebeccahongsf) [reply src/components/navigation/accountLinks.js:31]: Annd it seems that the search params survived without any additional URL encoding 😎

**Comment by rebeccahongsf** [comment]
Thank you for the speedy review @sherakama! 😃

File: src/components/auth/AuthenticatedPage.js [hunk 6]
```diff
@@ -1,10 +1,16 @@
...
 
 const AuthenticatedPage = ({ children, redirectUnauthorized = true }) => {
+  const location = useLocation();
+
   function redirectUnuauthorized() {
-    const returnUrl = window.location.pathname;
+    const returnUrl =
+      typeof window === 'undefined'
+        ? location.pathname
+        : window.location.pathname;
```
Comment (sherakama) [inline src/components/auth/AuthenticatedPage.js:13]: These will also need. `location.search` | `window.location.search`

**Pull request merged**